packages/salesforcedx-vscode-core/src/commands/isvdebugging/bootstrapCmd.ts

```diff
@@ -0,0 +1,189 @@
+    return new SfdxCommandBuilder()
+      .withDescription(
+        nls.localize('isv_debug_bootstrap_step1_configure_project')
+      )
+      .withArg('force:config:set')
+      .withArg(`isvDebuggerSid=${data.sessionId}`)
+      .withArg(`isvDebuggerUrl=${data.loginUrl}`)
+      .withArg(`instanceUrl=${data.loginUrl}`)
+      .build();
+  }
+
+  public execute(response: ContinueResponse<IsvDebugBootstrapConfig>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const projectParentPath = response.data.projectUri;
+    const projectPath = path.join(projectParentPath, response.data.projectName);
+
+    const createProjectExecution = new CliCommandExecutor(
+      this.buildCreateProjectCommand(response.data),
+      {
+        cwd: projectParentPath
+      }
+    ).execute(cancellationToken);
+
+    createProjectExecution.processExitSubject.subscribe(async data => {
+      if (data != undefined && data.toString() === '0') {
+        const configureProjectExecution = new CliCommandExecutor(
+          this.buildConfigureProjectCommand(response.data),
+          {
+            cwd: projectPath
+          }
+        ).execute(cancellationToken);
+
+        configureProjectExecution.processExitSubject.subscribe(async data2 => {
+          if (data2 != undefined && data2.toString() === '0') {
+            // last step is open the folder
+            await vscode.commands.executeCommand(
+              'vscode.openFolder',
+              vscode.Uri.parse(projectPath)
+            );
+          }
+        });
+
+        this.attachExecution(
+          configureProjectExecution,
+          cancellationTokenSource,
+          cancellationToken
+        );
+      }
+    });
+
+    this.attachExecution(
+      createProjectExecution,
+      cancellationTokenSource,
+      cancellationToken
+    );
+  }
+
+  protected attachExecution(
+    execution: CommandExecution,
+    cancellationTokenSource: vscode.CancellationTokenSource,
+    cancellationToken: vscode.CancellationToken
+  ) {
+    channelService.streamCommandOutput(execution);
+    channelService.showChannelOutput();
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    CancellableStatusBar.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+export type IsvDebugBootstrapConfig = ProjectNameAndPath & ForceIdeUri;
+
+export interface ForceIdeUri {
+  loginUrl: string;
+  sessionId: string;
+}
+
+export class EnterForceIdeUri implements ParametersGatherer<ForceIdeUri> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<ForceIdeUri>
+  > {
+    const forceIdeUrlInputOptions = {
+      prompt: nls.localize('parameter_gatherer_paste_forceide_url')
+    } as vscode.InputBoxOptions;
```

Comment:
Why do we need the explicit cast here? Is there something in the vscode.InputBoxOptions that we are not conforming to?

Reply:
It was a leftover. I simplified the code block.